src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java

```diff
@@ -93,140 +97,41 @@ public void launch(SlaveComputer computer, TaskListener listener) {
             return;
         }
 
-        KubernetesCloud cloud = slave.getKubernetesCloud();
-        final PodTemplate unwrappedTemplate = slave.getTemplate();
+        final PodTemplate template = slave.getTemplate();
         try {
-            KubernetesClient client = cloud.connect();
-            Pod pod = getPodTemplate(client, slave, unwrappedTemplate);
+            KubernetesClient client = slave.getKubernetesCloud().connect();
+            Pod pod = template.build(client, slave);
 
             String podId = pod.getMetadata().getName();
 
             String namespace = Arrays.asList( //
-                    pod.getMetadata() != null ? pod.getMetadata().getNamespace() : null,
-                    unwrappedTemplate.getNamespace(), client.getNamespace()) //
+                    pod.getMetadata().getNamespace(),
+                    template.getNamespace(), client.getNamespace()) //
                     .stream().filter(s -> StringUtils.isNotBlank(s)).findFirst().orElse(null);
             slave.setNamespace(namespace);
 
             LOGGER.log(Level.FINE, "Creating Pod: {0} in namespace {1}", new Object[]{podId, namespace});
             pod = client.pods().inNamespace(namespace).create(pod);
             LOGGER.log(INFO, "Created Pod: {0} in namespace {1}", new Object[]{podId, namespace});
-            logger.printf("Created Pod: %s in namespace %s%n", podId, namespace);
-
-            // We need the pod to be running and connected before returning
-            // otherwise this method keeps being called multiple times
-            List<String> validStates = ImmutableList.of("Running");
-
-            int waitForPodSec = cloud.getWaitForPodSec().intValue();
```

Comment:
This PR seems to have removed the one usage of a field introduced in #398 without removing the field itself.